src/network/changesManager/changesmanager.cpp

```diff
@@ -0,0 +1,75 @@
+#include "changesmanager.h"
+
+ChangesManager::ChangesManager()
+{
+
+}
+
+bool ChangesManager::getChange(const QString & newFile, Change & change, const SizeType fileId)
+{
+    QString currentFile = m_currentFiles.at(static_cast<int>(fileId));
+    // Find first & last mismatched element iterators
+    const auto firstMismatchedElement =
+            std::mismatch(currentFile.cbegin(), currentFile.cend(),
+                          newFile.cbegin());
+
+    if (firstMismatchedElement.first == currentFile.cend())
+    {
+        // Return if there is no changes
+        change.m_index  = 0;
+        change.m_before = "";
+        change.m_after  = "";
+        change.m_fileId = fileId;
+        return false;
+    }
+
+    const auto lastMismatchedElement =
+            std::mismatch(currentFile.crbegin(), currentFile.crend(),
+                          newFile.crbegin());
+    // Index of where has difference occured
+    SizeType occurenceIndex =
+            static_cast<SizeType>(std::distance(currentFile.cbegin(),
+                                                firstMismatchedElement.first));
+    // Two substrings that are different
+    QString substringBefore;
+    std::copy(firstMismatchedElement.first, lastMismatchedElement.first.base(), std::back_inserter(substringBefore));
```

Comment:
It will crash where the last mismatch element has lesser adress than the first mismatch element. For example, we have two strings "klu" and "k0lu". First mismatch iterator in the first string will be "lu"
 and the last mismatch iterator in the first string will be "k". "k" is lefward "lu".
Below, in the string 37 you're trying to create the string from range between these iterators:
std::copy(firstMismatchedElement.second, lastMismatchedElement.second.base(), std::back_inserter(substringAfter));	
Check it before creating string

Reply:
This is outdated file and is absent in context of build